test-case-reporting/src/db_interfacing.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2020 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Build, DB, Job, TestCase, TestRun} from 'test-case-reporting';
+import {Database, TIMESTAMP_PRECISION} from './db';
+
+const msConversionConstant: number = Math.pow(10, 3 - TIMESTAMP_PRECISION);
+
+// Types in the DB namespace interface use snake_case instead of camelCase.
+/* eslint @typescript-eslint/camelcase: "off" */
+function getDateFromTimestamp(timestamp: number): Date {
+  return new Date(timestamp * msConversionConstant);
+}
+
+export class TestResultsRecords {
+  constructor(private db: Database) {}
+
+  /**
+   * Gets a DB.Build from the database by its name.
+   * If there is no build, it returns undefined.
```

Comment:
It may be a good idea to throw an error instead of returning undefined -- what do you think? @rcebulko

Reply:
I'm inclined to use undefined since it's more like `myDict['missing key']` and simpler to handle than errors, but either is valid

Reply:
Re: (https://github.com/ampproject/amp-github-apps/pull/920#discussion_r450354029)
Okay, I'll leave it as it is.